bson-codec-design/DESIGN.md

```diff
@@ -0,0 +1,157 @@
+# BSON Encoder & Decoder Design
+The v1 design for the BSON library's encoder and decoder contain design issues that
+require a redesign. The main design flaw is the lack of a coherent, flexible design that can
+incorporate the needs of users. Assumptions made during the v1 design have proven incorrect or too
+inflexible to be as useful as initially intended.
+
+## Definitions
+<dl>
+<dt>Codec Processing</dt>
+<dd>The complete process of either encoding or decoding.</dd>
+<dt>Unmarshal</dt>
+<dd>The process of converting a slice of bytes into a native Go type</dd>
+<dt>Marshal</dt>
+<dd>The process of converting a native Go type into a slice of bytes</dd>
+<dt>Encode</dt>
+<dd>The process of converting a native Go type into BSON and writing it to an
+<code>io.Writer</code></dd>
+<dt>Decode</dt>
+<dd>The process of reading a BSON document from an <code>io.Reader</code> and converting it to a
+native Go type</dd>
+<dt>V1 Design </dt>
+<dd>The initial design for the encoding and decoding functionality of the BSON library</dd>
+<dt>Proposed Design</dt>
+<dd>The design laid out in this document</dd>
+</dl>
```

Comment:
is there any way to avoid in-lining HTML in markdown? 
Additionally, it would be nice to: 
- use complete sentences. 
- use a consistent capitalization for `v1`